hooks/pre-exit

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+echo "--- Cleaning up secrets"
+echo "Killing ssh-agent with PID ${SSH_AGENT_PID}"
+eval "$(ssh-agent -k)"
+echo "Removing secrets directory: ${SECRETS_DIR}"
```

Comment:
Shouldn't we expose this env via an environment hook in order for it to be accessible here?